packages/types/src/lib.ts

```diff
@@ -71,13 +72,21 @@ type Lib =
   | 'es2022.intl'
   | 'es2022.object'
   | 'es2022.regexp'
-  | 'es2022.sharedmemory'
```

Comment:
Newest typescript might have removed this, but the plugin should be compatible with previous TS versions (as per range in package.json) and those previous versions do know what `es2022.sharedmemory` is. This change makes this eslint plugin incompatible with previous TS versions in projects declaring use of `es2022.sharedmemory`

Reply:
Interesting. Could you please file an issue with a reproduction?

https://typescript-eslint.io/contributing/pull-requests

Reply:
Sure, on it

Reply:
https://github.com/typescript-eslint/typescript-eslint/issues/10395

Reply:
Actually, it's not this change, it's the rename of the lib file. Still caused by this PR